### DISCUSSION TITLE
refactor(bitbucket): replace accumulateValues with paginate getJson()

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#22272] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

With the introduction of https://github.com/renovatebot/renovate/pull/22135 should we raise a `type:refactor` issue to migrate / replace uses of `accumulateValues` function with the paginated getJson approach?

https://github.com/renovatebot/renovate/blob/f2977e4f55197eeb173638173bd1cb4eb5eb528d/lib/modules/platform/bitbucket/utils.ts#L100

**Comment by viceice** [comment]
yes, feel free to open a PR without an issue. 🤗

it would probably good to do that in multiple PR's, so we can review faster.

**Reply by setchy** [reply]
Absolutely. Will do

**Reply by HonkingGoose** [reply]
We got a PR:

- https://github.com/renovatebot/renovate/pull/22275

**Reply by setchy** [reply]
this needs to land first 😎 

- https://github.com/renovatebot/renovate/pull/22278

**Reply by setchy** [reply]
Here are the other refactor PRs.  once merged, there will be a final PR to remove the `accumulateValues` utility
- https://github.com/renovatebot/renovate/pull/22336
- https://github.com/renovatebot/renovate/pull/22330
- https://github.com/renovatebot/renovate/pull/22329
- https://github.com/renovatebot/renovate/pull/22348

**Reply by setchy** [reply]
Removal PR - https://github.com/renovatebot/renovate/pull/22359

**Comment by setchy** [comment]
This has now been completed
